### PR TITLE
Make all providers extend Provider, and fix HTTP service dependency types

### PR DESCRIPTION
**Pull request by rhyslbw**
# Context
Original implementations of some HTTP services were made depending on the concrete `DbSync*Provider` classes, rather than the interface. During review of #321, I noticed not all providers were extending `Provider`, leading to [this](https://github.com/input-output-hk/cardano-js-sdk/pull/321/files#diff-fd5402b8f766e8c72f9cceb402bae11b7c71cd91b904a2ad38954a4cfd869e9cR17)

# Proposed Solution
- Update all provider interfaces to extend `Provider`
- Replace concrete dependencies with equivalent interfaces

# Important Changes Introduced

### CODE REVIEW

File: packages/wallet/src/services/ProviderTracker/TrackedAssetProvider.ts [hunk 1]
```diff
@@ -19,12 +22,15 @@ export class AssetProviderStats {
  */
 export class TrackedAssetProvider extends ProviderTracker implements AssetProvider {
   readonly stats = new AssetProviderStats();
+  readonly healthCheck: AssetProvider['healthCheck'];
```
Comment (mkazlauskas) [inline packages/wallet/src/services/ProviderTracker/TrackedAssetProvider.ts:25]: Need to update tracked provider tests to include healthCheck

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

File: packages/wallet/src/services/ProviderTracker/TrackedAssetProvider.ts (hunk 1 above)
Comment (rhyslbw) [reply packages/wallet/src/services/ProviderTracker/TrackedAssetProvider.ts:25]: Done in https://github.com/input-output-hk/cardano-js-sdk/pull/346/commits/96d73416672a64fecb97bf345ad1a70ed0a48c48

**Review by iccicci — APPROVED** [review]
LGTM

**Review by iadmytro — DISMISSED** [review]

**Review by mkazlauskas — DISMISSED** [review]
Looks good!

**Review by mkazlauskas — APPROVED** [review]

**Pull request merged**